experimental/QuadFormAndIsom/src/embeddings.jl

```diff
@@ -364,37 +422,39 @@ function _fitting_isometries(OqfN::AutomorphismGroup{TorQuadModule},
   # _fHN is the coset _fN*KN where KN is the preimage by discrep of the kernel
   # of _imN. Now, inside this coset, some isometries could still give rise to
   # isomorphic equivariant primitive extensions for our classifying group. Thus
-  # we need to identity isometries which are conjugate by an isometry of our
+  # we need to identify isometries which are conjugate by an isometry of our
   # classifying group stabilizing HN (otherwise it does not make sense). This
-  # group of isometries is exactly the preimage of stabN by discrep, which we
-  # call CN here.
+  # group of isometries is exactly the preimage by discrep of the centralizer
+  # in stabN of _fN, which we call CN here.
   #
   # To summarize, in the general case, we obtain representatives of fitting
   # isometries by identifying CN-conjugate isometries in the coset fNKN.
   if first
     reporb = QQMatrix[solve(basis_matrix(N), basis_matrix(N)*matrix(_fN); side=:left)]
   else
     KNhat, _ = discrep\(kernel(_actN)[1])
+    _CN, _ = centralizer(_actN(stabN)[1], _imN(_fHN))
+    _CN, _ = _actN\_CN
+    CN, _ = discrep\_CN
+    @hassert :ZZLatWithIsom 3 is_normal_subgroup(KNhat, CN)
+    @hassert :ZZLatWithIsom 3 all(g -> g * _fN == _fN * g, gens(CN))
     fNKN = _fN*KNhat
```

Comment:
As far as I can see `_fN` is not defined in the body of the function. Is this code tested? It should error. Or am I missing something?

Reply:
You probably need to expand the code, it is defined in line 414. Note that the code is indeed tested and it does not seem to error.